core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -405,9 +405,7 @@ class SparkContext(config: SparkConf) extends Logging {
 
     _eventLogDir =
       if (isEventLogEnabled) {
-        val unresolvedDir = conf.get("spark.eventLog.dir", EventLoggingListener.DEFAULT_LOG_DIR)
-          .stripSuffix("/")
-        Some(Utils.resolveURI(unresolvedDir))
```

Comment:
I think in your case we need to percentile encode the `unresolvedDir` before calling `resolveURI`.